main.go

```diff
@@ -132,11 +136,22 @@ func realMain(ctx context.Context) error {
 	}()
 
 	// Wait for stop
-	select {
-	case err := <-errCh:
-		return fmt.Errorf("server error: %w", err)
-	case <-ctx.Done():
-		fmt.Fprint(os.Stderr, "\nserver is shutting down...\n")
+	if *flagServerUpMinutes != 0 {
+		timer := time.NewTimer(time.Duration(*flagServerUpMinutes) * time.Minute)
+		select {
+		case err := <-errCh:
+			return fmt.Errorf("server error: %w", err)
+		case <-timer.C:
```

Comment:
Since we're not actually cancelling the timer, it might be better to just use `<- time.After()` instead.

Reply:
done